rmw_fastrtps_shared_cpp/src/subscription.cpp

```diff
@@ -44,35 +44,23 @@ destroy_subscription(
   CustomParticipantInfo * participant_info,
   rmw_subscription_t * subscription)
 {
-  if (!subscription) {
-    RMW_SET_ERROR_MSG("subscription handle is null");
-    return RMW_RET_ERROR;
-  }
-  if (subscription->implementation_identifier != identifier) {
-    RMW_SET_ERROR_MSG("subscription handle not from this implementation");
-    return RMW_RET_ERROR;
-  }
-  if (!participant_info) {
-    RMW_SET_ERROR_MSG("participant_info is null");
-    return RMW_RET_ERROR;
-  }
+  assert(subscription->implementation_identifier == identifier);
```

Comment:
@hidmic In non-debug builds the `identifier` variable is now unused - see new compiler warnings: https://ci.ros2.org/view/packaging/job/packaging_linux/1969/gcc/

Reply:
Fixed in https://github.com/ros2/rmw_fastrtps/pull/422